src/modules/flow/iio/nodes.c

```diff
@@ -776,12 +782,24 @@ humidity_open(struct sol_flow_node *node, void *data, const struct sol_flow_node
 }
 
 static bool
-adc_create_channels(struct iio_double_data *mdata, int device_id)
+adc_create_channels(struct iio_double_data *mdata, int device_id, const char* channelname)
 {
+    char channel_name[NAME_MAX];
+    int ret;
+
     mdata->iio_base.device = sol_iio_open(device_id, &mdata->iio_base.config);
     SOL_NULL_CHECK(mdata->iio_base.device, false);
 
-    mdata->channel_val = iio_add_channel(mdata->scale, mdata->offset, "in_voltage0", &mdata->iio_base);
+    if (channelname) {
+        ret = snprintf(channel_name, sizeof(channel_name), "%s", channelname);
```

Comment:
@fulong82 "allow user input index number instead of the whole channel name" is the current solution in this PR. please update the webpage to get the latest PR #2318